### PR TITLE
CEDS- 1088 transport -details pages

### PR DESCRIPTION
**Pull request by raghu1978**
Split transport information page
create Journey based on user input
testing whole view 
pending - summary page
pending - transport details mapping to xml 
pending - view tests

### CODE REVIEW

File: app/controllers/declaration/BorderTransportController.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+)(implicit ec: ExecutionContext, appConfig: AppConfig,
+override val messagesApi: MessagesApi)
+    extends FrontendController with I18nSupport {
+
+  def form(): Form[BorderTransport] = Form(BorderTransport.formMapping)
+
+  def displayForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    customsCacheService
+      .fetchAndGetEntry[BorderTransport](cacheId, BorderTransport.formId)
+      .map ( data => Ok(border_transport(data.fold(form)(form.fill(_)))))
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/BorderTransportController.scala]: :+1: Yeah, better approach

File: app/controllers/declaration/BorderTransportController.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      .bindFromRequest()
+      .fold(
+        (formWithErrors: Form[BorderTransport]) =>
+          Future.successful(BadRequest(border_transport(formWithErrors))),
+        borderTransport =>
+          customsCacheService
+            .cache[BorderTransport](cacheId, BorderTransport.formId, borderTransport)
+            .map(_ => {
+              Redirect(TransportDetailsController.displayForm())
+            })
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/BorderTransportController.scala]: unnecessary `{ }`, Redirect should be in the same line like map

File: app/controllers/declaration/SealController.scala [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  def submitForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    val actionTypeOpt = request.body.asFormUrlEncoded.map(FormAction.fromUrlEncoded(_))
+
+    cacheService
+      .fetchAndGetEntry[Seq[Seal]](cacheId, formId)
+      .flatMap(cache => processRequest(cache.getOrElse(Seq.empty), actionTypeOpt))
+  }
+
+  private def processRequest(cachedSeals: Seq[Seal], action: Option[FormAction])(
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/SealController.scala]: I know that if the method is private you don't need to add return type, but here it will be saver to check that you receive what you want not something different that will be accepted.

Comment for whole file

File: app/controllers/declaration/SealController.scala [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  private def addSeal(boundForm: Form[Seal], elementLimit: Int, seals: Seq[Seal])(
+    implicit request: JourneyRequest[_],
+    appConfig: AppConfig
+  ) =
+    add(boundForm, seals, elementLimit).fold(
+      formWithErrors => Future.successful(BadRequest(seal(formWithErrors, seals))),
+      updatedCache => redirect(updatedCache)
+    )
+
+  private def redirect(seals: Seq[Seal])(implicit request: JourneyRequest[_]) = cacheService
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/SealController.scala]: Please change name to `cacheAndRedirect`

redirect is misleading

File: app/controllers/declaration/TransportDetailsController.scala [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          Future.successful(BadRequest(transport_details(formWithErrors))),
+        transportDetails =>
+          customsCacheService
+            .cache[TransportDetails](cacheId, TransportDetails.formId, transportDetails)
+            .map(_ => redirect(transportDetails))
+      )
+  }
+
+  private def redirect(transportDetails: TransportDetails)(implicit request: JourneyRequest[_]) =
+    true match {
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/TransportDetailsController.scala]: `true match`? How it can work? always will be summary page.

File: app/forms/declaration/BorderTransport.scala [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      text()
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.length", noLongerThan(27))
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.specialCharacters", isAlphanumeric)
+    )
+  )(BorderTransport.apply)(BorderTransport.unapply)
+}
+
+object TransportCodes {
+
+  val Maritime = "1"
```
Comment (PatrykRudnicki) [inline app/forms/declaration/BorderTransport.scala]: We have those codes in page that I added or around it.

File: app/forms/declaration/Seal.scala [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+object Seal  {
+  implicit val format = Json.format[Seal]
+
+  val formId = "Seal"
+
+  val formMapping = Forms.mapping(
+    "id" ->
+      text()
+        .verifying("standard.transport.sealId.empty.error", nonEmpty)
+        .verifying("standard.transport.sealId.alphaNumeric.error", isAlphanumeric)
```
Comment (PatrykRudnicki) [inline app/forms/declaration/Seal.scala]: Please change to `isEmpty or isAlphanumeric` because when input is empty we shouldn't show this error

File: app/services/Countries.scala [hunk 8]
```diff
@@ -56,5 +56,5 @@ object Countries {
 
   private def countryCode: String => String = cc => cc.split(":")(1).trim
 
-  val allCountries: List[Country] = countries.filter(c => mdgCountryCodes contains c.countryCode)
+  val allCountries: List[Country] = countries
```
Comment (PatrykRudnicki) [inline app/services/Countries.scala]: Why this change?

File: app/views/declaration/summary/summary_page.scala.html [hunk 9]
```diff
@@ -18,17 +18,24 @@
...
-@(suppDecData: SupplementaryDeclarationData)(implicit appConfig: AppConfig, request: Request[_], messages: Messages)
+@(suppDecData: SupplementaryDeclarationData)(implicit appConfig: AppConfig, request: JourneyRequest[_], messages: Messages)
 
 @main_template(
     title = messages("supplementary.summary.title"),
     appConfig = appConfig) {
 
-
-    @components.back_link("/customs-declare-exports/declaration/warehouse")
+    @{if(request.choice.value == "STD")
```
Comment (PatrykRudnicki) [inline app/views/declaration/summary/summary_page.scala.html]: Please import object and use variable for `"STD"` when the type will change or something like this it will be easier to maintain

File: app/views/declaration/summary/summary_page.scala.html [hunk 10]
```diff
@@ -18,17 +18,24 @@
...
+            components.back_link("/customs-declare-exports/declaration/add-seal")
+        else
```
Comment (PatrykRudnicki) [inline app/views/declaration/summary/summary_page.scala.html]: Please use parenthesis for else. It'll be more readable to understand what is inside

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+
+  val form: Form[BorderTransport] = Form(BorderTransport.formMapping)
+
+  def view(form: Form[BorderTransport] = form, request: FakeRequest[_]): Html =
+    border_transport(form)(request, messages, appConfig)
+
+  before {
+    authorizedUser()
+    withCaching[BorderTransport](None, BorderTransport.formId)
+    withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: Missing tests for standard dec? or this page is the same for both?

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+import uk.gov.hmrc.auth.core.InsufficientEnrolments
+import views.html.declaration.border_transport
+
+class BorderTransportControllerSpec extends CustomExportsBaseSpec with Generators with PropertyChecks {
+
+  private val uri = uriWithContextPath("/declaration/border-transport")
+
+  val form: Form[BorderTransport] = Form(BorderTransport.formMapping)
+
+  def view(form: Form[BorderTransport] = form, request: FakeRequest[_]): Html =
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: if you change order it will be easier to invoke it without specifying, so with this:
`def view(form: Form[BorderTransport] = form, request: FakeRequest[_])` you need to type `view(request = ....)`

but if you change order like this:

`def view(request: FakeRequest[_], form: Form[BorderTransport] = form)` you can use just `view(request)`

probably it's a good practice to put default value arguments at the end

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+    }
+  }
+
+  ".onSubmit" should {
+
+    "return UNAUTHORIZED" when {
+
+      "user does not have an EORI" in {
+        userWithoutEori()
+        withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: You specified this in before case

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+        intercept[InsufficientEnrolments](status(result))
+      }
+    }
+
+    "return BAD_REQUEST" when {
+
+      "invalid data is submitted" in {
+        authorizedUser()
+        withCaching[BorderTransport](None, BorderTransport.formId)
+        withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: You specified those two in before case

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    "add input to the cache" when {
+
+      "with valid data and on click of add" in {
+
+        forAll(arbitrary[BorderTransport]) { borderTransport =>
+          authorizedUser()
+          withCaching[BorderTransport](None, BorderTransport.formId)
+          withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: The same here

File: test/controllers/declaration/BorderTransportControllerSpec.scala [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+      }
+    }
+
+    "navigate to \"transport-details\" page" when {
+
+      "on click of continue when a record has already been added" in {
+        forAll(arbitrary[BorderTransport]) { borderTransport =>
+          authorizedUser()
+          withCaching[BorderTransport](Some(borderTransport), commodityFormId)
+          withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/BorderTransportControllerSpec.scala]: And here

File: test/controllers/declaration/SealControllerSpec.scala [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+import uk.gov.hmrc.auth.core.InsufficientEnrolments
+import views.html.declaration.seal
+
+class SealControllerSpec extends CustomExportsBaseSpec with Generators with PropertyChecks {
+
+  private val uri = uriWithContextPath("/declaration/add-seal")
+
+  val form: Form[Seal] = Form(Seal.formMapping)
+
+  def view(form: Form[Seal] = form, seals: Seq[Seal])(implicit request: FakeRequest[_]): Html =
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/SealControllerSpec.scala]: The same propose like in tests for `BorderTransportControllerSpec`

File: test/controllers/declaration/SealControllerSpec.scala [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+    }
+    "remove seal from the cache" when {
+
+      " user click remove" in {
+
+        forAll(listOf[Seal](sealArbitrary.arbitrary)) { (seals) =>
+          whenever(seals.size > 1) {
+            authorizedUser()
+            withCaching[Seq[Seal]](Some(seals), Seal.formId)
+            withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.StandardDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/SealControllerSpec.scala]: My propose is to use traits to define those mocks etc, it's easier to have a control on everything and to maintain the code. With this you can remove a lot of duplications

File: test/controllers/declaration/TransportDetailsControllerSpec.scala [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+import uk.gov.hmrc.auth.core.InsufficientEnrolments
+import views.html.declaration.transport_details
+
+class TransportDetailsControllerSpec extends CustomExportsBaseSpec with Generators with PropertyChecks {
+
+  private val uri = uriWithContextPath("/declaration/transport-details")
+
+  val form: Form[TransportDetails] = Form(TransportDetails.formMapping)
+
+  def view(form: Form[TransportDetails] = form, request: FakeRequest[_]): Html =
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/TransportDetailsControllerSpec.scala]: The same thing like before about arguments order

File: test/controllers/declaration/TransportDetailsControllerSpec.scala [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+    }
+  }
+
+  ".onSubmit" should {
+
+    "return UNAUTHORIZED" when {
+
+      "user does not have an EORI" in {
+        userWithoutEori()
+        withCaching[Choice](Some(Choice(Choice.AllowedChoiceValues.SupplementaryDec)), choiceId)
```
Comment (PatrykRudnicki) [inline test/controllers/declaration/TransportDetailsControllerSpec.scala]: Duplication of mock in all tests, please remove it

File: app/controllers/declaration/TransportDetailsController.scala (hunk 5 above)
Comment (raghu1978) [reply app/controllers/declaration/TransportDetailsController.scala]: transportDetails.container can be true
case seal if request.choice.value == AllowedChoiceValues.StandardDec can be true
 have tests for the same

File: app/forms/declaration/BorderTransport.scala (hunk 6 above)
Comment (raghu1978) [reply app/forms/declaration/BorderTransport.scala]: will change to have common codes

File: app/controllers/declaration/TransportDetailsController.scala (hunk 5 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/TransportDetailsController.scala]: Man, you're matching `true`... not transportDetails

File: app/services/Countries.scala (hunk 8 above)
Comment (raghu1978) [reply app/services/Countries.scala]: we do not have requirement to filter the countries and this is one of the reason some of the countries are not allowed.

File: app/forms/declaration/Seal.scala (hunk 7 above)
Comment (raghu1978) [reply app/forms/declaration/Seal.scala]: this error will not be shown id nonEmpty, I have test to prove

File: test/controllers/declaration/BorderTransportControllerSpec.scala (hunk 11 above)
Comment (raghu1978) [reply test/controllers/declaration/BorderTransportControllerSpec.scala]: this page is same for both

File: test/controllers/declaration/BorderTransportControllerSpec.scala (hunk 12 above)
Comment (raghu1978) [reply test/controllers/declaration/BorderTransportControllerSpec.scala]: there should never be a default form I will change

File: test/controllers/declaration/SealControllerSpec.scala (hunk 18 above)
Comment (raghu1978) [reply test/controllers/declaration/SealControllerSpec.scala]: I have updated to use in before and is much easier now

File: app/controllers/declaration/BorderTransportController.scala (hunk 2 above)
Comment (raghu1978) [reply app/controllers/declaration/BorderTransportController.scala]: updated

File: app/controllers/declaration/SealController.scala (hunk 4 above)
Comment (raghu1978) [reply app/controllers/declaration/SealController.scala]: updated

File: app/controllers/declaration/SealController.scala (hunk 3 above)
Comment (raghu1978) [reply app/controllers/declaration/SealController.scala]: updated

File: test/controllers/declaration/BorderTransportControllerSpec.scala (hunk 16 above)
Comment (raghu1978) [reply test/controllers/declaration/BorderTransportControllerSpec.scala]: updated

File: test/controllers/declaration/TransportDetailsControllerSpec.scala (hunk 20 above)
Comment (raghu1978) [reply test/controllers/declaration/TransportDetailsControllerSpec.scala]: updated

**Comment by raghu1978** [comment]
> Please use `scalafmt` the code in few places it's difficult to read and without one style, we will die with conflicts sometimes

File: test/controllers/declaration/SealControllerSpec.scala (hunk 17 above)
Comment (raghu1978) [reply test/controllers/declaration/SealControllerSpec.scala]: updated

File: test/controllers/declaration/TransportDetailsControllerSpec.scala (hunk 19 above)
Comment (raghu1978) [reply test/controllers/declaration/TransportDetailsControllerSpec.scala]: updated

File: app/controllers/declaration/TransportDetailsController.scala (hunk 5 above)
Comment (raghu1978) [reply app/controllers/declaration/TransportDetailsController.scala]: that is what I had and updated that to. this to be more readable

File: app/controllers/declaration/BorderTransportController.scala [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+
+class BorderTransportController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  errorHandler: ErrorHandler,
+  customsCacheService: CustomsCacheService
+)(implicit ec: ExecutionContext, appConfig: AppConfig, override val messagesApi: MessagesApi)
+    extends FrontendController with I18nSupport {
+
+  def form(): Form[BorderTransport] = Form(BorderTransport.formMapping)
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/BorderTransportController.scala]: In my opinion form should be inside companion object

File: app/controllers/declaration/SealController.scala [hunk 22]
```diff
@@ -0,0 +1,114 @@
...
+  override val messagesApi: MessagesApi,
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  errorHandler: ErrorHandler,
+  cacheService: CustomsCacheService
+)(implicit ec: ExecutionContext, appConfig: AppConfig)
+    extends FrontendController with I18nSupport {
+
+  def form(): Form[Seal] = Form(formMapping)
+  val sealsAllowed = 9999
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/SealController.scala]: The same here, I'll move it to Seal object

File: app/controllers/declaration/TransportDetailsController.scala [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+  authenticate: AuthAction,
+  journeyAction: JourneyAction,
+  errorHandler: ErrorHandler,
+  customsCacheService: CustomsCacheService
+)(implicit ec: ExecutionContext, appConfig: AppConfig, override val messagesApi: MessagesApi)
+    extends FrontendController with I18nSupport {
+
+  implicit val countries = services.Countries.allCountries
+
+  def form(): Form[TransportDetails] = Form(TransportDetails.formMapping)
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/TransportDetailsController.scala]: The same here about form. Should be in TransportDetails companion object

File: app/forms/declaration/BorderTransport.scala [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
```
Comment (PatrykRudnicki) [inline app/forms/declaration/BorderTransport.scala]: Still I'm thinking that this will be duplication with Warehouse page. I didn't see rest of the PR yet, so I don't know that you removed it from different place or not.

File: app/forms/declaration/TransportInformation.scala [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+package forms.declaration
+
+import forms.MetadataPropertiesConvertable
+import play.api.data.Forms.{boolean, optional, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+import services.Countries.allCountries
+import utils.validators.forms.FieldValidator._
+
+case class TransportInformation(
```
Comment (PatrykRudnicki) [inline app/forms/declaration/TransportInformation.scala]: Do we need need this class still?

File: app/forms/declaration/TransportInformation.scala [hunk 26]
```diff
@@ -0,0 +1,162 @@
...
+
+object TransportInformation {
+  implicit val format = Json.format[TransportInformation]
+
+  val id = "TransportInformation"
+
+  def form(): Form[TransportInformation] = Form(mapping)
+
+  import ModeOfTransportCodes._
+  private val allowedModeOfTransportCodes =
```
Comment (PatrykRudnicki) [inline app/forms/declaration/TransportInformation.scala]: This allowedModeOfTransportCodes you have in object TransportCodes

File: app/forms/declaration/TransportInformation.scala [hunk 27]
```diff
@@ -0,0 +1,162 @@
...
+    Set(Maritime, Rail, Road, Air, PostalConsignment, FixedTransportInstallations, InlandWaterway, Unknown)
+
+  import MeansOfTransportTypeCodes._
+  private val allowedMeansOfTransportTypeCodes =
```
Comment (PatrykRudnicki) [inline app/forms/declaration/TransportInformation.scala]: The same with this

File: app/forms/declaration/TransportInformation.scala [hunk 28]
```diff
@@ -0,0 +1,162 @@
...
+      text()
+        .verifying(
+          "supplementary.transportInfo.meansOfTransport.crossingTheBorder.nationality.error.incorrect",
+          isContainedIn(allCountries.map(_.countryName))
+        )
+    ),
+    "container" -> boolean
+  )(TransportInformation.apply)(TransportInformation.unapply)
+
+  object ModeOfTransportCodes {
```
Comment (PatrykRudnicki) [inline app/forms/declaration/TransportInformation.scala]: This ModeOfTransportCodes too. In TransportCodes

File: app/forms/declaration/TransportInformation.scala [hunk 29]
```diff
@@ -0,0 +1,162 @@
...
+    val Rail = "2"
+    val Road = "3"
+    val Air = "4"
+    val PostalConsignment = "5"
+    val FixedTransportInstallations = "7"
+    val InlandWaterway = "8"
+    val Unknown = "9"
+  }
+
+  object MeansOfTransportTypeCodes {
```
Comment (PatrykRudnicki) [inline app/forms/declaration/TransportInformation.scala]: This too

File: app/models/declaration/TransportInformationContainerData.scala [hunk 30]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+
+package models.declaration
+
+import forms.MetadataPropertiesConvertable
+import forms.declaration.TransportInformationContainer
+import play.api.libs.json.Json
+
+case class TransportInformationContainerData(containers: Seq[TransportInformationContainer])
```
Comment (PatrykRudnicki) [inline app/models/declaration/TransportInformationContainerData.scala]: So we're coming back to old solution with model for view and model for cache?

File: app/services/WcoMetadataMapping.scala [hunk 31]
```diff
@@ -15,31 +15,96 @@
...
 package services
-import forms.declaration.{Document, PreviousDocumentsData}
+import forms.declaration._
 import models.DeclarationFormats._
 import models.declaration.SupplementaryDeclarationData
+import services.Countries.allCountries
 import uk.gov.hmrc.http.cache.client.CacheMap
-import uk.gov.hmrc.wco.dec.{Declaration, GovernmentAgencyGoodsItem, MetaData, PreviousDocument}
-
+import uk.gov.hmrc.wco.dec.{
```
Comment (PatrykRudnicki) [inline app/services/WcoMetadataMapping.scala]: Hmm, I'm thinking about import with `_`

`import uk.gov.hmrc.wco.dec._`

File: app/views/declaration/transport_details.scala.html [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+
+    @helpers.form(TransportDetailsController.submitForm(), 'autoComplete -> "off") {
+
+        @components.error_summary(form.errors)
+
+        @components.page_title(Some("supplementary.transportInfo.title"), None)
+      
+         @components.input_text_autocomplete(form)(
+            field = form("meansOfTransportCrossingTheBorderNationality"),
+            label = messages("supplementary.transportInfo.meansOfTransport.crossingTheBorder.nationality.header"),
```
Comment (PatrykRudnicki) [inline app/views/declaration/transport_details.scala.html]: And for the future, my propose is to make this message keys shorter

File: app/controllers/declaration/BorderTransportController.scala (hunk 21 above)
Comment (raghu1978) [reply app/controllers/declaration/BorderTransportController.scala]: don't provide any more than moving the mapping

File: app/forms/declaration/BorderTransport.scala (hunk 24 above)
Comment (raghu1978) [reply app/forms/declaration/BorderTransport.scala]: it has been removed

File: app/forms/declaration/TransportInformation.scala (hunk 25 above)
Comment (raghu1978) [reply app/forms/declaration/TransportInformation.scala]: waiting for summary page updates

File: app/forms/declaration/TransportInformation.scala (hunk 26 above)
Comment (raghu1978) [reply app/forms/declaration/TransportInformation.scala]: this class will be removed later

File: app/forms/declaration/TransportInformation.scala (hunk 28 above)
Comment (raghu1978) [reply app/forms/declaration/TransportInformation.scala]: the codes are all sorted don't worry about it

File: app/services/WcoMetadataMapping.scala (hunk 31 above)
Comment (raghu1978) [reply app/services/WcoMetadataMapping.scala]: that would be too big of a package

File: app/views/declaration/transport_details.scala.html (hunk 32 above)
Comment (raghu1978) [reply app/views/declaration/transport_details.scala.html]: not my code we have to move these later

File: app/controllers/declaration/SealController.scala (hunk 22 above)
Comment (raghu1978) [reply app/controllers/declaration/SealController.scala]: done

File: app/controllers/declaration/BorderTransportController.scala (hunk 21 above)
Comment (raghu1978) [reply app/controllers/declaration/BorderTransportController.scala]: changed

File: app/controllers/declaration/TransportDetailsController.scala (hunk 23 above)
Comment (raghu1978) [reply app/controllers/declaration/TransportDetailsController.scala]: updated

File: app/forms/declaration/TransportInformation.scala (hunk 25 above)
Comment (raghu1978) [reply app/forms/declaration/TransportInformation.scala]: leaving it to look into how we can handle summary page

File: app/models/declaration/TransportInformationContainerData.scala (hunk 30 above)
Comment (raghu1978) [reply app/models/declaration/TransportInformationContainerData.scala]: not yet  I am leaving with what's built already and removing the ones that are being added are built new
